File: src/main/java/ru/citeck/ecos/index/indexers/qname/QName.java
package ru.citeck.ecos.index.indexers.qname;

import lombok.Data;

@Data
public class QName {

    public static final String QNAME_PATTERN = "org.alfresco.service.namespace.QNamePattern";
    public static final String CLASS = "org.alfresco.service.namespace.QName";

    private final String localName;
    private final String uri;
    private final String prefix;
    private final String javaField;
    private final String javaClass;

}

File: src/main/kotlin/ru/citeck/alfresco/QName.kt
package ru.citeck.alfresco

class QName(
    val localName: String,
    val uri: String,
    val jField: String,
    val jClass: String
) {

    companion object {
        const val CLASS = "org.alfresco.service.namespace.QName"
    }

    override fun toString(): String {
        return "${uri ?: ""}:$localName $jClass.$jField"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as QName

        if (localName != other.localName) return false
        if (uri != other.uri) return false
        if (jField != other.jField) return false
        if (jClass != other.jClass) return false

        return true
    }

    override fun hashCode(): Int {
        var result = localName.hashCode()
        result = 31 * result + uri.hashCode()
        result = 31 * result + jField.hashCode()
        result = 31 * result + jClass.hashCode()
        return result
    }

}